src/miner/worker.rs

```diff
@@ -114,7 +113,7 @@ impl Worker {
             let challenge_nonce = Worker::start(&channel, update, &mut rng);
             challenge = challenge_nonce.0;
             nonce = challenge_nonce.1;
-            is_first = true;
+            current_difficulty = challenge.difficulty.saturating_sub(1);
```

Comment:
I see the `current_difficulty` variable being updated here and on line 106 but I don't see the variable being used anywhere else afterwards so I'm not sure what to make of it.

Reply:
I didn't see the loop on line 90.